run.py

```diff
@@ -0,0 +1,18 @@
+import logging.config
+from apscheduler.schedulers.background import BackgroundScheduler
+from time import sleep
+from janitor.tasks.labware_location.main import sync_changes_from_labwhere
+from janitor.helpers.config_helpers import get_config
+from dotenv import load_dotenv
+
+load_dotenv()
+
+config = get_config()
+logging.config.dictConfig(config.LOGGING)
+
+if __name__ == "__main__":
+    sched = BackgroundScheduler(daemon=False)
+    sched.add_job(sync_changes_from_labwhere, "interval", [config], seconds=config.SYNC_JOB_INTERVAL_SEC)
+    sched.start()
+    while True:
+        sleep(300)
```

Comment:
I agree this Stephen. Does setting daemon=True achieve this ^. The docs aren't clear on whether this will keep it running after main has executed but it sounds like it should. It may mean you need a shutdown script to kill the processes when you want to stop it though.

Reply:
So the reason we had this infinite loop was to keep the thread alive for the background scheduler. Without this, the scheduler would just start and as soon as that line is executed, the program would exit. `daemon=True` unfortunately did not stop it from terminating.

Some sort of loop has to be present for any of the schedulers to run because this script will be running 'forever' in the Docker container. It will be the only script running and we don't plan on shutting it down unless the container itself gets shut down, in which case the script would be force stopped.

However, we can use `BlockingScheduler` instead of `BackgroundScheduler` and this will allow us to achieve the same thing but without having to create our own loop. The thread being kept alive is handled internally but it is still just doing an infinite loop until we want to stop the tasks. I will go with this as it does tidy up our code and also provides us with a graceful way of shutting down, even though not strictly necessary